### DISCUSSION TITLE
glb conversion - preserve hierarchy?

### DISCUSSION THREAD

**Discussion by lrwillis** [IfcOpenShell/IfcOpenShell#3725] [Q&A]
When converting from IFC to GLB, is there a way to preserve the IFC's hierarchical structure?  (e.g. Project>Site>Building>Building Storey>Walls/etc.)

When I look at the output of the default IFC->GLB conversion, the deep hierarchical structure of the IFC file appears to have been flattened. 

Thanks,
Lee

**Comment by lrwillis** [comment]
That attached picture shows how the structure was flattenned.  On the left is a simple house model.   You can see that the staircase is nested several layers deep in the hierarchy structure.  

On the right, is the GLB conversion of the same file.   Here the staircase is right under the root node.

![Ifc_to_GLB_hierarchy](https://github.com/IfcOpenShell/IfcOpenShell/assets/79932400/c584146e-5c7a-41cd-9142-776ca2a97bc9)

**Comment by aothms** [comment]
One thing you can try is to use IfcConvert to collada (`.dae`) and then use one of the various tools to convert that to gltf. When you convert to `.dae` you have the `--use-element-hierarchy` option, which was unfortunately specifically implemented in the collada serializer so is not available in other serializers. (all help welcome btw :))
